# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1208983915 14400
#      Wed Apr 23 16:51:55 2008 -0400
# Branch release-3-0-x
# Node ID 8e6dc8dd94e742a07ce5af18c0347877c9b1f092
# Parent  6a8fd365eebd29ace49c6c86f1be67471467d087
__qp__.cc: avoid bounds error when removing constraint from active set

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2008-04-23  John W. Eaton  <jwe@octave.org>
+
+	* DLD-FUNCTIONS/__qp__.cc (qp): Avoid bounds error when removing
+	constraint from active set.
+
 2008-04-21  John W. Eaton  <jwe@octave.org>
 
 	Version 3.0.1 released.
 
 	* version.h (OCTAVE_VERSION): Now 3.0.1.
 	(OCTAVE_RELEASE_DATE): Now 2008-04-21.
 	(OCTAVE_COPYRIGHT): Update year.
 
diff --git a/src/DLD-FUNCTIONS/__qp__.cc b/src/DLD-FUNCTIONS/__qp__.cc
--- a/src/DLD-FUNCTIONS/__qp__.cc
+++ b/src/DLD-FUNCTIONS/__qp__.cc
@@ -350,24 +350,24 @@ qp (const Matrix& H, const ColumnVector&
 			  which_eig = i;
 			  break;
 			}
 		    }
 
 		  // At least one multiplier is negative, we
 		  // remove it from the set.
 
+		  n_act--;
 		  for (octave_idx_type i = which_eig; i < n_act - n_eq; i++)
 		    {
 		      Wact(i) = Wact(i+1);
 		      for (octave_idx_type j = 0; j < n; j++)
 			Aact(n_eq+i,j) = Aact(n_eq+i+1,j);
 		      bact(n_eq+i) = bact(n_eq+i+1);
 		    }
-		  n_act--;
 
 		  // Resizing the active set.
 		  Wact.resize (n_act-n_eq);
 		  bact.resize (n_act);
 		  Aact.resize (n_act, n);
 		}
 	    }
 	}
